File: SIMINOV-CONNECT-TEMPLATE/assets/template/services/GetLiquorBrands.java
package siminov.connect.template.services;

import java.util.Iterator;

import siminov.connect.design.connection.IConnectionRequest;
import siminov.connect.design.connection.IConnectionResponse;
import siminov.connect.exception.ServiceException;
import siminov.connect.service.Service;
import siminov.connect.template.fragments.LiquorDetail;
import siminov.connect.template.model.Liquor;
import siminov.connect.template.model.LiquorBrand;
import siminov.connect.template.reader.LiquorBrandsReader;
import siminov.orm.exception.DatabaseException;
import siminov.orm.log.Log;
import android.support.v4.app.Fragment;

public class GetLiquorBrands extends Service {

	public static final String SERVICE_NAME = "SIMINOV-CONNECT-LIQUOR-BRANDS";
	public static final String API_NAME = "GET-LIQUOR-BRANDS";
	
	public static final String LIQUOR_NAME = "LIQUOR-NAME";
	public static final String LIQUOR = "LIQUOR";
	
	private Liquor liquor = null;
	private Fragment fragment = null;
	
	
	public GetLiquorBrands() {
		setService(SERVICE_NAME);
		setApi(API_NAME);
	}

	public Fragment getComponent() {
		return this.fragment;
	}
	
	public void setComponent(Fragment fragment) {
		this.fragment = fragment;
	}
	
	public Liquor getLiquor() {
		return this.liquor;
	}
	
	public void setLiquor(Liquor liquor) {
		this.liquor = liquor;
	}
	
	public void onServiceStart() {

	}

	public void onServiceQueue() {
		
	}

	public void onServicePause() {
		
	}

	public void onServiceResume() {
			
	}

	public void onServiceFinish() {
		
	}

	public void onServiceApiInvoke(IConnectionRequest connectionRequest) {
		
	}

	public void onServiceApiFinish(IConnectionResponse connectionResponse) {
		
		LiquorBrandsReader liquorBrandsReader = new LiquorBrandsReader(connectionResponse.getResponse());
		Iterator<LiquorBrand> liquorBrands = liquorBrandsReader.getLiquorBrands();
		
		while(liquorBrands.hasNext()) {
			LiquorBrand liquorBrand = liquorBrands.next();
			liquorBrand.setLiquor(liquor);
			
			try {
				liquorBrand.saveOrUpdate();
			} catch(DatabaseException de) {
				Log.loge(GetLiquors.class.getName(), "onServiceApiFinish", "Database Exception caught while saving liquor brands in database, " + de.getMessage());
			}
		}
		
		((LiquorDetail) fragment).refresh();
	}

	public void onServiceTerminate(ServiceException serviceException) {
		
	}
}

File: connect-sample/iOS/Siminov/Connect/Sample/Services/GetLiquorBrands.m
//
//  GetLiquorBrands.m
//  connect-sample
//
//  Created by Geetika on 29/05/15.
//  Copyright (c) 2015 Siminov. All rights reserved.
//

#import "GetLiquorBrands.h"
#import "SIKIConnectionResponse.h"
#import "Liquor.h"
#import "LiquorBrandsReader.h"

static NSString * const SERVICE_NAME = @"SIMINOV-CONNECT-LIQUOR-BRANDS-SERVICE";
static NSString * const REQUEST_NAME = @"GET-LIQUOR-BRANDS";

static NSString * const LIQUOR_NAME = @"LIQUOR_NAME";
static NSString * const LIQUOR = @"LIQUOR";

@implementation GetLiquorBrands

- (id)init {
    
    if (self = [super init]) {
        [self setService:SERVICE_NAME];
        [self setRequest:REQUEST_NAME];
    }
    return self;
}

- (void)onStart {
    
}

- (void)onQueue {
    
}

- (void)onPause {
    
}

- (void)onResume {
    
}

- (void)onFinish {
    
}

- (void)onRequestInvoke:(id<SIKIConnectionRequest>)connectionRequest {
}

- (void)onRequestFinish:(id<SIKIConnectionResponse>)connectionResponse {
    
    if ([connectionResponse getResponse] == nil) {
        return;
    }
    
    Liquor *liquor = [self getResource:@"LIQUOR"];
    LiquorBrandsReader *liquorBrandsReader = [[LiquorBrandsReader alloc]initWithData:[connectionResponse getResponse]];
    NSEnumerator *liquorBrands = [liquorBrandsReader getLiquorBrands];
    
    
    LiquorBrand *liquorBrand;
    while (liquorBrand = [liquorBrands nextObject]) {
        [liquorBrand setLiquor:liquor];
        
        @try {
            [liquorBrand saveOrUpdate];
        }
        @catch (SICDatabaseException *exception) {
            [SICLog error:NSStringFromClass([self class]) methodName:@"onServiceApiFinish" message:[NSString stringWithFormat:@"Database Exception caught while saving liquor brands in database, %@",[exception reason]]];
        }
    }
    //((LiquorDetail) uiComponent).refresh();
}

- (void)onTerminate:(SIKServiceException *)serviceException {
}

@end
